### ISSUE TITLE
browser() jumps around when code is added

### ISSUE THREAD

**Issue by billdenney** [rstudio/rstudio#10061]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2021.09.0 (build 351)
    OS Version      : Windows 10
    R Version       : 4.1.2

### Steps to reproduce the problem

1. Source a script with a `browser()` call in a function (rename this to .R, for example: [file1.txt](https://github.com/rstudio/rstudio/files/7505174/file1.txt))
2. Run the function (`my_browser()`)
3. The debug browser will jump to the correct location.
4. Add code below the `browser()` call in the function, and press Ctrl-Enter to run that new code in the console.
5. A new `code_browser://` window will pop up (see the second code tab in the attached image): ![image](https://user-images.githubusercontent.com/10359931/140936534-25d9dfbd-8261-4558-b34f-838d7b69541f.png)
6. Move back to the original code tab (file1.R in my case), and repeat steps 4-6.

Code modification for step 4, I just added `x <- 1`.

```r
my_browser <- function() {
  browser()
  x <- 1
}
```

### Describe the problem in detail

This may be related to #10011 (but, I'm not sure)

Jumping to the `code_browser://` tab is only helpful when you are not already at the correct location in the code.

### Describe the behavior you expected

It would be helpful if a new `code_browser://` tab only popped up when a browser call just occurred and not after pressing ctrl-enter.  My preference for behavior occurs when the function is not modified, but does not happen when the function is modified.

In other words, if you do not modify the function and run code within the function by pressing ctrl-enter it does not jump to a `code_browser://` window.  And, if you do not modify the function and paste code into the console, it does not jump to a `code_browser://` window.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by Robinlovelace** [comment]
+1 to that, it's really annoying when trying to refactor and debug functions.

**Comment by kevinushey** [comment]
Do you also see this in the latest daily builds of RStudio? https://dailies.rstudio.com/

**Comment by Robinlovelace** [comment]
> Do you also see this in the latest daily builds of RStudio? https://dailies.rstudio.com/

Not tried, will do so on my laptop when I get a chance. Thanks!

**Comment by Robinlovelace** [comment]
Tried the daily on laptop. First thing that happened when trying to build a package for local testing was this. Possibly reproducible example: open a project containing a package and press Ctl+Shift+B

![image](https://user-images.githubusercontent.com/1825120/142169757-3bae5c83-7ca1-409f-acb6-176e78bbab71.png)

**Comment by Robinlovelace** [comment]
Update, after opening the project a second time eventually (there seems to be a bug but I lack time to report it sorry), I cannot reproduce the error. Happy to say: I think this is fixed upstream, as illustrated below.

![image](https://user-images.githubusercontent.com/1825120/142170200-333c6e89-32f7-4ed2-a824-3275e85e7798.png)

**Comment by billdenney** [comment]
> Do you also see this in the latest daily builds of RStudio? https://dailies.rstudio.com/

I just tried it out, and it does not exhibit the problem with jumping around.  Thanks!

From my perspective, this can be closed (but I don't know if there is more that needs to happen from the Rstudio side of things, so I will leave it to @kevinushey ).

**Comment by billdenney** [comment]
As I've been using the daily build a bit more, there appears to be higher CPU usage and possibly generally slower use of Rstudio.  I don't know if this is related, and I can't immediately figure out a reprex for it.  But, it may be good to look at CPU usage while in the `browser()`.

**Comment by billdenney** [comment]
This appears to be fixed in the current stable version.
